prompting-client/src/daemon/worker.rs

```diff
@@ -202,8 +209,13 @@ where
 
         loop {
             match self.wait_for_expected_prompt(&expected_id).await {
-                Recv::Success | Recv::Timeout | Recv::Gone => break,
                 Recv::DeadPrompt | Recv::Unexpected => continue,
+                Recv::Success | Recv::Gone => break,
+                Recv::Timeout => {
+                    let reply = prompt.into_deny_once();
+                    self.client.reply(&expected_id, reply).await?;
+                    break;
+                }
```

Comment:
What is the origin of the `Recv::Timeout`? Is this a timeout defined in the client server, or something received from the prompting API request?

Reply:
Ah sorry, this is internal within the prompting client itself as part of [wait_for_expected_prompt](https://github.com/canonical/prompting-client/blob/main/prompting-client/src/daemon/worker.rs#L232) in the worker thread. The happy path flow is that the UI will respond to the GRPC server and wait for an ack that its reply has been successfully sent to snapd before closing. We don't attempt to check that the prompt has been actioned from the worker until after we see the UI exit, so the timeout here is around the worker receiving a message from the GRPC server that snapd accepted the reply so it can begin processing the next prompt.